### PR TITLE
examples: Add Java code showing object upload to Ceph RGW bucket

### PR DESCRIPTION
**Pull request by faithuniterh**
Signed-off-by: FaithUniterh <faithuniterh@tutanota.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: examples/java/ceph-s3-upload/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Introduction
+This directory contains Java code examples on how to upload an object to an S3 bucket running on a Ceph RGW cluster.
+
+# Prerequisite
+Linux machine running a RGW Ceph cluster with Java and Maven installed.
+
+## Workflow Procedure
+1. Install AWS CLI version one on your Linux machine as explained [here](https://docs.aws.amazon.com/cli/v1/userguide/install-linux.html)
```
Comment (anthonyeleven) [inline examples/java/ceph-s3-upload/README.md:8]: s/one/1/

File: examples/java/ceph-s3-upload/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+2. Once successful, run ```aws configure ``` on your terminal to enter the Ceph RGW cluster access key and secret key into the global shared configuration file.
+3. Create a bucket on the Ceph cluster with the command 
+```
+aws --endpoint-url http://localhost:8000 s3 mb s3://sample-bucket
+```
+4. Navigate through your file system into the ceph-s3-upload folder  using your terminal. Please ensure you see the pom.xml file.
```
Comment (anthonyeleven) [inline examples/java/ceph-s3-upload/README.md:14]: Wrap `ceph-s3-upload` in double backticks.  The triple backticks elsewhere here should probably be double just for consistency.

File: examples/java/ceph-s3-upload/ceph-s3-upload.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+This file will be uploaded to ceph S3 storage
```
Comment (anthonyeleven) [inline examples/java/ceph-s3-upload/ceph-s3-upload.txt:1]: Capitalize Ceph.

Comment (faithuniterh) [reply examples/java/ceph-s3-upload/ceph-s3-upload.txt]: Done

File: examples/java/ceph-s3-upload/README.md (hunk 2 above)
Comment (faithuniterh) [reply examples/java/ceph-s3-upload/README.md]: All done

**Comment by faithuniterh** [comment]
@yuvalif Please have a look at this when you get time. Thank you.

File: examples/java/ceph-s3-upload/README.md (hunk 1 above)
Comment (anthonyeleven) [reply examples/java/ceph-s3-upload/README.md]: I'm suggesting that you change "version one" to "version 1"

**Comment by yuvalif** [comment]
same issue with credentials as in here: https://github.com/ceph/ceph/pull/45937#discussion_r857125657

File: examples/java/ceph-s3-upload/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Introduction
+This directory contains Java code examples on how to upload an object to an S3 bucket running on a Ceph RGW cluster.
+
+# Prerequisites
+Linux machine running an RGW Ceph cluster. Preferrably started with the ``RGW=1 ../src/vstart.sh --debug --new -x`` command.
```
Comment (yuvalif) [inline examples/java/ceph-s3-upload/README.md:5]:
```suggestion
Linux machine running an RGW Ceph cluster. Preferrably started with the ``OSD=1 MON=1 MDS=0 MGR=0 RGW=1 ../src/vstart.sh --debug --new`` command.  
```

**Review by yuvalif — APPROVED** [review]

**Comment by yuvalif** [comment]
* please update README
* please squash commits

**Review by yuvalif — APPROVED** [review]

**Pull request merged**